### DISCUSSION TITLE
SimpleStorage" is unknown import symbol

### DISCUSSION THREAD

**Discussion by fricpto** [smartcontractkit/full-blockchain-solidity-course-py#1744] [Q&A] [answered]
I have imported SimpleSorage and I have this error message:
"SimpleStorage" is unknown import symbol
and this is my code:
from eth_account import account
from brownie import accounts, config,  SimpleStorage
def deploy_simple_storage():
    account = accounts[0]
simple_storage =  SimpleStorage.deploy({"from": account})
print(simple_storage)

<img width="429" alt="Screenshot 2022-10-14 235443" src="https://user-images.githubusercontent.com/100000316/195942384-995ba9f7-3ec9-476e-b5e7-63c5f9c3e864.png">

def main():
deploy_simple_storage()

**Comment by cromewar** [comment, marked as answer]
Hello @fricpto 
Please be sure you have the contract on the correct folder path, delete the build folder and compile again before importing.
Finally be sure to select the correct python interpreter on vscode terminal.
Hope this info might help.

**Reply by Niferu** [reply]
Hello @cromewar @fricpto 
I have just encountered same error. No idea why because it was working fine for me and now it seems like pylance doesnt sees any of my imports. It is caused by interpreter for sure, so pathing might be incorrect, but after hour i still cannot fix it no idea what happened that it stopped seeing my imports I have removed PyCharm from my PC before that happened but I doubt it caused the error. Any help on this will be appreciated. Will let you know once I find soluction.

**Reply by Niferu** [reply]
@cromewar solution with deleting `build` folder then compiling works, but I still cannot get rid of that error below, which drives me crazy as it was all good until today... 

![image](https://user-images.githubusercontent.com/110537615/196326461-64dd4f9e-7918-4665-a78f-77b7ff023de3.png)

It even started throwing errors as below:
![image](https://user-images.githubusercontent.com/110537615/196326928-1cdc0404-64db-4443-843b-4ba6c897d2ac.png)


Any idea how to fix this? It throws errors for every import from brownie

**Reply by Niferu** [reply]
Ok I have reinstalled literally everything including VSCode and now my setup is completely broken. There is still same error occuring and my brownie is pointing on folders which does not exists and I'm keep getting `[PermissionError: [Errno 13] Permission denied]` errors... I have wasted like 4 hours yesterday and 5 hours today to fix it and it just got worse I'm just totally woeful

**Reply by cromewar** [reply]
Hello @Neftyr 
Go to your home folder (~ in linux/mac or C/users/yourUser on Windows) you will find a hidden folder called `.brownie` delete it.
Secondly, which python interpreter are you using? depending on the interpreter it would have permissions to access one or more paths, so...
Is better to use a python virtual env (python 3.8 preferably) and install brownie on it.

**Reply by Niferu** [reply]
Hi @cromewar 
I have installed python 3.10.6 as everything was working perfectly fine until yesterday night... I have already deleted that `.brownie` folder as I wanted to get rid of all files to fix that annyoing error and now I can open directory with `.brownie` but I cannot see it anymore in C/users/yourUser. I have also tested bunch of other interpreters but nothing worked.

Omg @cromewar sorry I'm dumb I have ticked `hidden` option for `.brownie` file now it works and brownie performing my actions. So the problem is still with that imports by Pylance as it throws tons of those... any idea how to fix that? I have to add that my code is working now but not sure what caused Pylance to start acting like that.

another pylance error example:
![image](https://user-images.githubusercontent.com/110537615/196447865-de21ae84-3830-4eb8-a16f-5332177446d7.png)

**Reply by fricpto** [reply]
@cromewar  thanks for the help but I have tried your advice but it didn't help.

**Reply by fricpto** [reply]
@cromewar So it is better to use python 3.8  and install brownie on it it will fix my problem?

**Reply by Niferu** [reply]
Hi @fricpto 
I have tried it and it didnt fix it (I have installed like 10 different versions of python), spent like 11 hours total trying to solve that but nothing helped... I have reinstalled python, all packages, vscode etc nothing worked. But despite your problem my code is working but some keywords are highlighted anyway by `pylance`. Searched tons of solutions over google but nothing helped. It was working all good for me when suddenly errors popped idk how is that even possible...

**Reply by fricpto** [reply]
@Neftyr my code is not working

**Reply by Niferu** [reply]
@fricpto 
It was not working for me too when this nightmare started, but I have just deleted that 1 build folder and recompiled brownie and it worked for all my project I have ever made, no idea how 1 build delete fixed all projects but it is the same oposite how doing nothing caused me this error...

**Reply by fricpto** [reply]
@Neftyr every time I delete the build folder and then I compile it reapers again.

**Reply by Niferu** [reply]
@fricpto and @cromewar 

This is some kind of joke, I DID NOTHING opened my VSCode now and all errors GONE how is this possible??? I was restarting PC and VSCode 100000 times and nothing now it worked wtf... Maybe auto VSCode update crashed it for me and now it is fixed somehow idk wasted 11 hours jesus...

@fricpto 
Send me screen of your whole code, maybe I will be able to help you somehow as i need to see your syntax highlighting

**Reply by fricpto** [reply]
@Neftyr 
<img width="328" alt="Screenshot 2022-10-19 000337" src="https://user-images.githubusercontent.com/100000316/196544453-e8b9b259-7c90-4ad1-8bef-e89e1bc98e09.png">

**Reply by cromewar** [reply]
@fricpto 
Be sure the name on the file on vscode for your contract is exactly `SimpleStorage` and also the contract itself (on the code) should be called that way, may be you have a typo here.
And yes the build folder will appear again, this is automatically generated when you compile the contract.

**Reply by Niferu** [reply]
@fricpto 
Yes follow cromewar advice as it should be the case, because your interpreted and path looks correct as it highlights everything correctly.

**Reply by fricpto** [reply]
@cromewar @Neftyr I just checked  if I have a type error but I don't have.  what should I do?

**Reply by Niferu** [reply]
Hi @fricpto 
Please send us code of your contract SimpleStorage.sol

**Reply by cromewar** [reply]
Yes please.

**Reply by fricpto** [reply]
@cromewar @Neftyr this is my SimpleStorage.sol code


```Solidity
// SPDX-License-Identifier: MIT

pragma solidity >=0.6.0 <0.9.0;

contract SimpleStorage {
    uint256 favoriteNumber;

    // This is a comment!
    struct People {
        uint256 favoriteNumber;
        string name;
    }

    People[] public people;
    mapping(string => uint256) public nameToFavoriteNumber;

    function store(uint256 _favoriteNumber) public {
        favoriteNumber = _favoriteNumber;
    }

    function retrieve() public view returns (uint256) {
        return favoriteNumber;
    }

    function addPerson(string memory _name, uint256 _favoriteNumber) public {
        people.push(People(_favoriteNumber, _name));
        nameToFavoriteNumber[_name] = _favoriteNumber;
    }
}
```
<img width="188" alt="Screenshot 2022-10-20 133322" src="https://user-images.githubusercontent.com/100000316/196925992-4b2df9a3-7f70-4354-9459-261d9ca2d6c7.png">

**Reply by Niferu** [reply]
Hi @fricpto 
To post your code correctly you should start with below

{```}solidity

paste your code here

{```}

Just remove all `{}`

Your contract looks good. This has to be smth wrong with interpreter and it is almost the same case as I had problem with but only exception here is that I could run my code and it was working even with same error you have. As even if you put incorrect contract name in your `deploy.py` it should not be highlighted red as it is for you. Pylance just does not know correct path to brownie in my opinion but I'm still learning and I have no idea how to fix path for pylance, so maybe cromewar will be albo to help here.. I have still no idea how it crashed for me out of nowhere and then repaired itself after 2 days...

Ahh and 1 more question to you @fricpto , was it working for you earlier? I'm not sure why you have `node_module` folder added to your paths on left, have you started this project with `brownie init`? Because it looks like front end setup with those `.json` files and this might be the case here...

Try to:
1. make new project so in your console `cd ..` you should be in Demos folder in your console, then `mkdir xxx` then `cd xxx` and at the end `code .`
2. As you are now in new window and project in console write `brownie init` then you should have below folder structure:
![image](https://user-images.githubusercontent.com/110537615/196943481-50c6364f-902c-43e6-9e8d-c5bc0c905d4c.png)
3. Recreate your `SimpleStorage.sol` file and `deploy.py`, then check if problem is solved.

Also make sure if you have correct interpreter at the bottom right it should look like this:
![image](https://user-images.githubusercontent.com/110537615/196943992-acd4fd69-131e-49c4-998a-f0310c77bb05.png)
And once you click on it there should be your python
![image](https://user-images.githubusercontent.com/110537615/196944233-534bbdd8-9a6e-47fa-b98b-5366d63dddc0.png)

**Reply by cromewar** [reply]
Incredible support @Neftyr 
I couldn't said it better.
Additionally @fricpto, could you please clone the repo and try to run it, does it run it properly?

**Reply by fricpto** [reply]
@Neftyr thanks for your help the error disappeared but the code still isn't working my interpreter is 3.10.7 64 -bit. this what happens  when I compile it gives me this message in the pic
<img width="424" alt="Screenshot 2022-10-20 205305" src="https://user-images.githubusercontent.com/100000316/197022377-909f3d72-e623-4eba-8679-98eab152b232.png">
and this is my code:
``` python
from eth_account import account
from brownie import accounts, config, SimpleStorage 
def deploy_simple_storage():
    account = accounts[0]
simple_storage = SimpleStorage.deploy({"from": account})
print(simple_storage)


def main():
    deploy_simple_storage()
```
should I delete python and then reinstalled?

**Reply by Niferu** [reply]
Hi @fricpto 
It is not about python i believe as your error disappeared. First of all your 1st line of code should be removed as you are not using `eth_account` + your line `simple_storage = SimpleStorage.deploy({"from": account})` and `print(simple_storage)` should be 1 tab ahead so in line with `account = accounts[0]`

**Reply by fricpto** [reply]
@cromewar I cloned the repo and it worke=

**Reply by fricpto** [reply]
@Neftyr Thanks a lot for your help the code actually worked without error

**Reply by cromewar** [reply]
I'm glad is solved now, I'll mark this as solved.
